libs/dkg/include/dkg/dkg_messages.hpp

```diff
@@ -0,0 +1,234 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/byte_array/const_byte_array.hpp"
+#include "core/serializers/byte_array.hpp"
+#include "core/serializers/byte_array_buffer.hpp"
+#include "network/muddle/rpc/client.hpp"
+
+namespace fetch {
+namespace dkg {
+
+using DKGSerializer = fetch::serializers::ByteArrayBuffer;
+
+/**
+ * Different messages using in distributed key generation (DKG) protocol.
+ * Coefficients - contain the broadcast coefficients as strings
+ * Shares - contain the secret shares which have been exposed in broadcasts as strings
+ * Complaints - contain the set of miners who are being complained against
+ */
+
+class DKGMessage
+{
+public:
+  using MsgSignature  = byte_array::ConstByteArray;
+  using MuddleAddress = byte_array::ConstByteArray;
+  using Coefficient   = std::string;
+  using Share         = std::string;
+  using CabinetId     = MuddleAddress;
+
+  enum class MessageType : uint8_t
+  {
+    COEFFICIENT,
+    SHARE,
+    COMPLAINT
+  };
+
+  virtual ~DKGMessage() = default;
+
+  /// @name Getter functions
+  /// @{
+  MessageType type() const
+  {
+    return type_;
+  }
+  MsgSignature signature() const
+  {
+    return signature_;
+  }
+  /// @}
+
+  virtual DKGSerializer Serialize() const = 0;
+
+protected:
+  const MessageType type_;       ///< Type of message of the three listed above
+  MsgSignature      signature_;  ///< ECDSA signature of message
+
+  explicit DKGMessage(MessageType type)
+    : type_{type} {};
```

Comment:
Note: a semicolon is unnecessary after a function body (including a ctor). It's just an empty statement.

Reply:
Thanks. I have fixed this the latest commit.